Add constant serialized size trait

paberr opened the pull request:
As per wish of @gakonst, here is a convenience trait that allows to get the serialized size for constant sized elements without requiring an instance of the element.

This also includes a commit that adds proper derivation of the uncompressed variant.
So far, derived serialization always used the compressed variant.

paberr committed:
Add constant serialized size trait

Pratyush commented:
If the output is always constant, maybe it makes sense to make it an associated const?

paberr committed:
Properly derive uncompressed (de-)serialization

paberr commented:
> If the output is always constant, maybe it makes sense to make it an associated const?

Good idea! I changed it and also added proper derivation of the uncompressed variants.
So far, the derivation didn't implement the uncompressed variant at all (falling back to the compressed).

Pratyush commented:
Awesome, thanks!
